src/server/engine_shard_set.cc

```diff
@@ -710,18 +698,18 @@ void EngineShard::RunPeriodic(std::chrono::milliseconds period_ms) {
 }
 
 void EngineShard::CacheStats() {
-  if (!namespaces.IsInitialized()) {
-    return;
-  }
-
```

Comment:
This was added here because the periodic heartbeat can be called after `namespaces` was cleared (happens on shutdown). Below we still access `namespaces.GetDefaultNamespace()`, which is not allowed.
Am I missing something and this is fine now?

Reply:
oh, I thought it was added because of the starting phase ( we had a data race there as well)

Reply:
I do not see how shutdown can cause a race.  this is the sequence:
```
  shard_set->PreShutdown();
  namespaces.Clear();
  shard_set->Shutdown();
```
PreShutdown stops the periodic fiber.

Reply:
Ok, you're right. It has a potential race before I split it, but I forgot about this.
My bad.